code/modules/research/rdconsole.dm

```diff
@@ -196,438 +196,445 @@ won't update every console in existence) but it's more of a hassle to do. Also,
+			submenu = next_submenu
 
-		matching_designs.Cut()
+		if("setCategory")
+			var/next_category = params["category"]
+			var/compare
 
-		if(menu == 4)
-			compare = PROTOLATHE
-		else
-			compare = IMPRINTER
+			matching_designs.Cut()
 
-		for(var/v in files.known_designs)
-			var/datum/design/D = files.known_designs[v]
-			if(!(D.build_type & compare))
-				continue
-			if(href_list["category"] in D.category)
-				matching_designs.Add(D)
-		submenu = 1
-
-		selected_category = "Viewing Category [href_list["category"]]"
-
-	else if(href_list["updt_tech"]) //Update the research holder with information from the technology disk.
-		add_wait_message("Updating Database...", TECH_UPDATE_DELAY)
-		spawn(TECH_UPDATE_DELAY)
-			clear_wait_message()
-			files.AddTech2Known(t_disk.stored)
-			SSnanoui.update_uis(src)
-			griefProtection() //Update centcom too
-
-	else if(href_list["clear_tech"]) //Erase data on the technology disk.
-		if(t_disk)
-			t_disk.wipe_tech()
-
-	else if(href_list["eject_tech"]) //Eject the technology disk.
-		if(t_disk)
-			t_disk.loc = src.loc
-			t_disk = null
-		menu = 0
-		submenu = 0
-
-	else if(href_list["copy_tech"]) //Copy some technology data from the research holder to the disk.
-		// Somehow this href makes me very nervous
-		t_disk.stored = files.known_tech[href_list["copy_tech_ID"]]
-		menu = 2
-		submenu = 0
-
-	else if(href_list["updt_design"]) //Updates the research holder with design data from the design disk.
-		add_wait_message("Updating Database...", DESIGN_UPDATE_DELAY)
-		spawn(DESIGN_UPDATE_DELAY)
-			clear_wait_message()
-			files.AddDesign2Known(d_disk.blueprint)
-			SSnanoui.update_uis(src)
-			griefProtection() //Update centcom too
-
-	else if(href_list["clear_design"]) //Erases data on the design disk.
-		if(d_disk)
-			d_disk.wipe_blueprint()
-
-	else if(href_list["eject_design"]) //Eject the design disk.
-		if(d_disk)
-			d_disk.loc = src.loc
-			d_disk = null
-		menu = 0
-		submenu = 0
-
-	else if(href_list["copy_design"]) //Copy design data from the research holder to the design disk.
-		// This href ALSO makes me very nervous
-		var/datum/design/D = files.known_designs[href_list["copy_design_ID"]]
-		if(D)
-			// eeeeeep design datums are global be careful!
-			var/autolathe_friendly = 1
-			for(var/x in D.materials)
-				if( !(x in list(MAT_METAL, MAT_GLASS)))
+			if(menu == 4)
+				compare = PROTOLATHE
+			else
+				compare = IMPRINTER
+
+			for(var/v in files.known_designs)
+				var/datum/design/D = files.known_designs[v]
+				if(!(D.build_type & compare))
+					continue
+				if(next_category in D.category)
+					matching_designs.Add(D)
+			submenu = 1
+
+			selected_category = "Viewing Category [next_category]"
+
+		if("updt_tech") //Update the research holder with information from the technology disk.
+			add_wait_message("Updating Database...", TECH_UPDATE_DELAY)
+			spawn(TECH_UPDATE_DELAY)
+				clear_wait_message()
+				files.AddTech2Known(t_disk.stored)
+				SStgui.update_uis(src)
```

Comment:
So returning TRUE from tgui_act makes an update happen, but since this is in a spawn I have to call `SStgui.update_uis(src)` right?

Reply:
Yes, to clear the waiting for sync modal (assuming you did that)